### PR TITLE
GH-13 Wrapping Adobe I/O Events Publishing API

### PR DESCRIPTION
**Pull request by francoisledroff**
added the Publish API wrapper service

### CODE REVIEW

File: events_ingress/src/main/java/com/adobe/event/publish/model/CloudEvent.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+package com.adobe.event.publish.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.util.Objects;
+import java.util.UUID;
+import org.apache.commons.lang3.StringUtils;
+
+public class CloudEvent {
```
Comment (sangeetha5491) [inline events_ingress/src/main/java/com/adobe/event/publish/model/CloudEvent.java:21]: We can use the Cloud event java SDK instead. https://github.com/cloudevents/sdk-java

Comment (francoisledroff) [reply events_ingress/src/main/java/com/adobe/event/publish/model/CloudEvent.java:21]: yes I guess we could ... but that would require quite a bit of work
let me add this to the project backlog for now: https://github.com/adobe/aio-lib-java/issues/16

File: events_ingress/src/main/java/com/adobe/event/publish/PublishServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+class PublishServiceImpl implements PublishService {
+
+  private final PublishApi publishApi;
+
+  PublishServiceImpl(final RequestInterceptor authInterceptor,
+      final String url) {
+    String apiUrl = StringUtils.isEmpty(url) ? PublishApi.DEFAULT_URL : url;
+    if (authInterceptor == null) {
+      throw new IllegalArgumentException(
+          "RegistrationService is missing a authentication interceptor");
```
Comment (sangeetha5491) [inline events_ingress/src/main/java/com/adobe/event/publish/PublishServiceImpl.java:32]: should be publish service is missing ....

File: events_ingress/src/main/java/com/adobe/event/publish/api/PublishApi.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  String DEFAULT_URL = "https://eventsingress.adobe.io";
+
+  /**
+   * publish a Cloud Event
+   *
+   * @param body your CloudEvent Input Model
+   */
+  @RequestLine("POST ")
+  @Headers({"Content-Type: application/cloudevents+json"})
+  void post(CloudEvent body);
```
Comment (sangeetha5491) [inline events_ingress/src/main/java/com/adobe/event/publish/api/PublishApi.java:30]: maybe we can rename it to publish?

File: events_ingress/src/main/java/com/adobe/event/publish/api/PublishApi.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+public interface PublishApi {
+
+  String DEFAULT_URL = "https://eventsingress.adobe.io";
+
+  /**
+   * publish a Cloud Event
+   *
+   * @param body your CloudEvent Input Model
+   */
+  @RequestLine("POST ")
```
Comment (sangeetha5491) [inline events_ingress/src/main/java/com/adobe/event/publish/api/PublishApi.java:28]: there is a space here. not sure if its intentional "POST"

File: events_ingress/src/main/java/com/adobe/event/publish/PublishServiceImpl.java (hunk 2 above)
Comment (francoisledroff) [reply events_ingress/src/main/java/com/adobe/event/publish/PublishServiceImpl.java]: done

File: events_ingress/src/main/java/com/adobe/event/publish/api/PublishApi.java (hunk 3 above)
Comment (francoisledroff) [reply events_ingress/src/main/java/com/adobe/event/publish/api/PublishApi.java]: done

File: events_ingress/src/main/java/com/adobe/event/publish/api/PublishApi.java (hunk 4 above)
Comment (francoisledroff) [reply events_ingress/src/main/java/com/adobe/event/publish/api/PublishApi.java]: yes extra space was not harmful but not needed either

**Review by sangeetha5491 — APPROVED** [review]

**Pull request merged**